### PR TITLE
build(nix): add nix builds and devshell

### PR DESCRIPTION
**Pull request by yusdacra**

### CODE REVIEW

**Comment by Songtronix** [comment]
Can it be checked in CI to be working?

**Comment by yusdacra** [comment]
> Can it be checked in CI to be working?

I added a `nix-test` job to `checks.yml`. I'm not sure if the Git LFS steps are required, but the Test job had it so I added them here as well (since it will also run tests).

**Comment by Songtronix** [comment]
> 
> 
> > Can it be checked in CI to be working?
> 
> I added a `nix-test` job to `checks.yml`. I'm not sure if the Git LFS steps are required, but the Test job had it so I added them here as well (since it will also run tests).

As lfs traffic even counts for CI I would remove it as long as it isn't needed for the tests otherwise just leave it. Gonna trust you that everything else just works ^^

File: .github/workflows/check.yml [hunk 1]
```diff
@@ -11,6 +11,54 @@ on:
...
+        extra_nix_config: |
+          experimental-features = nix-command flakes
+        nix_path: nixpkgs=channel:nixos-unstable
+
+    # Setup Cachix cache
+    - name: Setup cachix
+      uses: cachix/cachix-action@v10
+      with:
+        name: airshipper
+        authToken: '${{ secrets.CACHIX_AUTH_TOKEN }}'
```
Comment (Songtronix) [inline .github/workflows/check.yml:56]: Do I have to supply this secret?

Comment (yusdacra) [reply .github/workflows/check.yml:56]: I will create a token for that cache specifically, I can send it over Discord if you want? Or if you add me to project I can add it myself.

Comment (Songtronix) [reply .github/workflows/check.yml:56]: CI succeeded without this token. Looks like this is a "global cache"? Interesting especially security wise but I assume you know what's up ^^ You can send me the token in the meantime.

**Comment by yusdacra** [comment]
Oh also I should note that this does not add any NixOS specific patching for downloaded executables. I plan to do that in another PR.

**Review by Songtronix — APPROVED** [review]

**Comment by Songtronix** [comment]
> Oh also I should note that this does not add any NixOS specific patching for downloaded executables. I plan to do that in another PR.

Yeah probably wait for the 0.5 release as a lot of things will move around and change. Furthermore what kind of patches is it? And by hand or some kind of nix tool which takes care of that?

**Comment by yusdacra** [comment]
> > Oh also I should note that this does not add any NixOS specific patching for downloaded executables. I plan to do that in another PR.
> 
> Yeah probably wait for the 0.5 release as a lot of things will move around and change. Furthermore what kind of patches is it? And by hand or some kind of nix tool which takes care of that?

The `veloren-voxygen` library will need to be patched with https://github.com/NixOS/patchelf to have it's RPATH and dynamic loader point to correct `/nix/store` paths. Airshipper will need to call `nix` and have it evaluate an expression to do these fix-up stuff. I guess it would be nice if it was shown in the UI although they should not take more than a few seconds (at least with subsequent runs).

**Comment by Songtronix** [comment]
Okay ready to get merged?

**Comment by yusdacra** [comment]
> Okay ready to get merged (to staging of course)?

Pushed one last flake dependency update, should be good to go now.

**Pull request merged**